test/e2e/subscription_e2e_test.go

```diff
@@ -2813,48 +2818,58 @@ func checkDeploymentWithPodConfiguration(t GinkgoTInterface, client operatorclie
 		return
 	}
 
-	check := func(container *corev1.Container) {
+	check := func(container *corev1.Container) error {
 		for _, e := range envVar {
 			existing, found := findEnvVar(container.Env, e.Name)
-			require.Truef(t, found, "env variable name=%s not injected", e.Name)
-			require.NotNil(t, existing)
-			require.Equalf(t, e.Value, existing.Value, "env variable value does not match %s=%s", e.Name, e.Value)
+			if !found || existing == nil {
+				return errors.New(fmt.Sprintf("env variable name=%s not injected", e.Name))
```

Comment:
we can simply return a formatted error string through `return fmt.Errorf("env variable name=%s not injected", e.Name)`.

Reply:
Thanks!  I'll update.